### ISSUE TITLE
Inconsistent analog speed

### ISSUE THREAD

**Issue by DMXrated01** [simple64/simple64#378]
I have noticed that whenever I'm using the analog stick, whatever or whoever I am controlling moves at its proper speed when I hold it diagonally, but slower when I hold it vertically or horizontally.

To verify that something was wrong, I tested it in the beginning of Super Mario 64, and noticed that even when I have Deadzone and Analog Sensitivity set to 0% and 120% respectively, Mario moves much faster when I hold the stick diagonally than straight.

https://www.youtube.com/watch?v=UMcyW5QCjvM

For my target game, Getter Love!! Panda Love Unit, I went on to settle for the speed I had the analog on, but then found myself unable to move as quickly during minigames as when I used to play my hard copy.

Might be subtle, but notice how much slower my character (red) moves in some directions than the others.

https://youtu.be/0Wv71AFvr7k

Using:

Windows 10
Version 22H2
GPU: AMD Radeon Vega 8 Graphics

**Comment by loganmc10** [comment]
This does indeed need to be fixed, but I believe it's actually the opposite. In simple64, the diagonal directions are too fast, and left/right/up/down are correct. I'll look into correcting it

**Comment by DMXrated01** [comment]
Well, actually, the cardinal directions are too slow.  Just yesterday, I tested the stick on a different emulator, RMG Portable, where I set analog sensitivity to 175%, and saw Mario to move forward at full speed.  Back in Getter Love!!, I was able to move the map cursor at a comfortable speed, and was able to move at full speed during a minigame.

I just woke up right now, so I will create a new video later to demonstrate such differences if you need me to.

**Comment by DMXrated01** [comment]
Well, here it is.  I've just recorded a comparison in SM64 on RMG Portable.

Notice how much faster he moves when I set analog sensitivity back to 175% from 100%.

I think implementing a higher AS cap like that might solve the problem here.

<iframe width="560" height="315" src="https://www.youtube.com/embed/4Q8DNaSIzWk" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>

**Comment by loganmc10** [comment]
Yeah but this isn't an RMG emulator, there are ROMs to test controller behavior, for instance: https://github.com/sanni/controllertest/tree/master/N64-Port

I'll use that to try and get the behaviour closer to a real N64, but I'm not trying to mimic the behaviour of RMG

**Comment by DMXrated01** [comment]
Well, as long as I got my point across.

Good luck.

**Comment by loganmc10** [comment]
By the way, is this using a controller, or the keyboard? If controller, which controller?

**Comment by DMXrated01** [comment]
Using a standard Nintendo controller.  Is there any information I should look up about it?

**Comment by loganmc10** [comment]
Can you run the attached test ROM and post a screenshot of the results for your controller? There is a test in that ROM where it has you move the controller in each direction and then shows the results, it's this test:

![image](https://github.com/simple64/simple64/assets/848146/5f03790d-a711-4a54-be92-0c7526b5f9bf)
[controllertest.zip](https://github.com/simple64/simple64/files/12156615/controllertest.zip)

**Comment by DMXrated01** [comment]
Right now, it is not feasible.  Just the other day, I learned through Microsoft tech support that my computer had been deeply infected, and I would assume that from all those emulators I had downloaded to test my game on.

My brother will be here next month, and I will be more than happy to fill him in on things and get his help developing a safe approach to this.

**Comment by DMXrated01** [comment]
Sorry to move the goalposts on you like this, but my brother had just come down with coronavirus, and has postponed his visit for the middle of September.

Currently, I am planning to buy an extra computer, once I receive my broken one back all fixed up.  I do not wish to do that earlier, because if he were to find out, he might misinterpret me to have given up on him prematurely, and write off the one I had just mailed back to him.

(I had also attempted to take care of this at my library not too long ago, but due to certain complications, I was not able to access this thread through the email link I've been using.)
